stellargraph/data/unsupervisedSampler.py

```diff
@@ -0,0 +1,229 @@
+    """
+
+    def __init__(
+        self, G, walker=None, nodes=None, length=2, number_of_walks=1, seed=None
+    ):
+
+        """
+           Initializing all the paramters for the class instance. A Stellargraph is a must. 
+           The root nodes and walker are required. However, if not provided they default to all nodes in the graph and uniform random walker respectively.
+           The walks should have length at least 2. Need 2 nodes for a valid taget,contect pair.
+           At least one walk should be performed from each root node. Actually number of walks can be completely ignored here. Since in the end we just need the batch size to generate samples. 
+           So if for example, is large, we will do as many walks as needed to generate enough samples. The reason this parameter is still needed is to estimate how many batches in each epoch
+           are generated. 
+           
+           Args:
+               G <StellarGraph>: A stellargraph is a must.
+               walker <UniformRandomWalker>: The only walking method supported for sampling is Uniform Random Walk.
+               nodes <iterable of nodes from the graph>: if none provided, defaults to all nodes in the graph as root nodes for walks.
+               length <int>: An integer value for the length of the walk. Minimum length is 2 since the samples are a pair of target,context so at least one hop is needed beyond the root node.
+               number_of_walks <int>: Number of walks from each root node. Note, at the moment this parameter is only used to calculate the number of batches in each epoch,
+                                      as we do not have a pregenerated sample list, we need to estimate given a batch_size, how many batches to generate in each epoch.  
+               
+        
+        """
+        # Initialize the random state
+        self._random_state = random.Random(seed)
+
+        # Initialize a numpy random state (for numpy random methods)
+        self._np_random_state = np.random.RandomState(seed=seed)
+
+        if not isinstance(G, StellarGraphBase):
+            raise ValueError(
+                "({}) Graph must be a StellarGraph object.".format(type(self).__name__)
+            )
+        else:
+            self.graph = G
+
+        # walker
+        if walker is not None:
+            if not isinstance(
+                walker, UniformRandomWalk
+            ):  # only work with Uniform Random Walker at the moment
+                raise TypeError(
+                    "({}) Only Uniform Random Walks are possible".format(
+                        type(self).__name__
+                    )
+                )
+            else:
+                self.walker = UniformRandomWalk(G)
+        else:
+            self.walker = UniformRandomWalk(G)
+
+        if nodes is None:
+            self.nodes = (
+                G.nodes()
+            )  # if no root nodes are provided for sampling defaulting to using all nodes as root nodes.
```

Comment:
It might be good to make `self.nodes` a list

Reply:
actually a NodeView might be better in terms of memory footprint. Turning it into a list "unwraps" the NodeView into the actual list, while NodeView is just an iterator, right?

Reply:
I have changed this to a list so that we can shuffle the nodes before every set of walks.